### ISSUE TITLE
[ICE]: Missing value for constant, but no error reported?

### ISSUE THREAD

**Issue by cdrksm** [rust-lang/rust#154556]
### Code

Slightly reduced code can be found here: https://github.com/cdrksm/minimal-ice

### Meta

`rustc --version --verbose`:
```
rustc 1.96.0-nightly (fb27476aa 2026-03-28)
binary: rustc
commit-hash: fb27476aaf1012f1f6ace6306f9b990e0d989c31
commit-date: 2026-03-28
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.2
```

### Error output

```
Caused by:
  process didn't exit successfully: `/home/$USER/.local/share/rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustc --crate-name tes_plugin4 --edition=2024 tes-plugin4/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=212 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -Z codegen-backend=cranelift -C debuginfo=2 --check-cfg 'cfg(docsrs,test)' --check-cfg 'cfg(feature, values())' -C metadata=51311448e734f4ca -C extra-filename=-cdce788163fb60de --out-dir /home/$USER/src/rust/bsa/target/debug/deps -C incremental=/home/$USER/src/rust/bsa/target/debug/incremental -L dependency=/home/$USER/src/rust/bsa/target/debug/deps --extern bethesda_types=/home/$USER/src/rust/bsa/target/debug/deps/libbethesda_types-c1bd3ff48f35acf1.rmeta --extern encoding_rs=/home/$USER/src/rust/bsa/target/debug/deps/libencoding_rs-d18754dc19b3efa9.rmeta --extern tracing=/home/$USER/src/rust/bsa/target/debug/deps/libtracing-523dd9bbaedd6df0.rmeta -C link-arg=-fuse-ld=mold` (exit status: 101)
```

<details><summary><strong>Backtrace</strong></summary>
<p>
<details><summary>Debug build (cranelift)</summary>
<pre>
delayed bug: Missing value for constant, but no error reported?
   0: <rustc_errors::DiagCtxtInner>::emit_diagnostic
   1: <rustc_errors::DiagCtxtHandle>::emit_diagnostic
   2: <rustc_span::ErrorGuaranteed as rustc_errors::diagnostic::EmissionGuarantee>::emit_producing_guarantee
   3: <rustc_errors::DiagCtxtHandle>::span_delayed_bug::<rustc_span::span_encoding::Span, &str>
   4: rustc_trait_selection::traits::const_evaluatable::is_const_evaluatable.cold
   5: <rustc_trait_selection::traits::select::SelectionContext>::evaluate_root_obligation
   6: rustc_traits::evaluate_obligation::evaluate_obligation
   7: rustc_query_impl::query_impl::evaluate_obligation::invoke_provider_fn::__rust_begin_short_backtrace
   8: rustc_query_impl::execution::try_execute_query::<rustc_middle::query::caches::DefaultCache<rustc_type_ir::canonical::CanonicalQueryInput<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::ParamEnvAnd<rustc_middle::ty::predicate::Predicate>>, rustc_middle::query::erase::ErasedData<[u8; 2]>>, true>
   9: rustc_query_impl::query_impl::evaluate_obligation::execute_query_incr::__rust_end_short_backtrace
  10: <rustc_infer::infer::InferCtxt as rustc_trait_selection::traits::query::evaluate_obligation::InferCtxtExt>::evaluate_obligation_no_overflow
  11: <core::iter::adapters::map::Map<core::iter::adapters::filter::Filter<core::slice::iter::Iter<rustc_hir_typeck::method::probe::Candidate>, <rustc_hir_typeck::method::probe::ProbeContext>::consider_candidates::{closure#0}>, <rustc_hir_typeck::method::probe::ProbeContext>::consider_candidates::{closure#1}> as core::iter::traits::iterator::Iterator>::try_fold::<(), core::iter::traits::iterator::Iterator::find::check<(&rustc_hir_typeck::method::probe::Candidate, rustc_hir_typeck::method::probe::ProbeResult), &mut <rustc_hir_typeck::method::probe::ProbeContext>::consider_candidates::{closure#2}>::{closure#0}, core::ops::control_flow::ControlFlow<(&rustc_hir_typeck::method::probe::Candidate, rustc_hir_typeck::method::probe::ProbeResult)>>
  12: <rustc_hir_typeck::method::probe::ProbeContext>::pick_method
  13: <rustc_hir_typeck::method::probe::ProbeContext>::pick_all_method
  14: <rustc_hir_typeck::fn_ctxt::FnCtxt>::lookup_probe
  15: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_method_call
  16: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  17: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_overloaded_binop
  18: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_binop
  19: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  20: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_block
  21: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  22: rustc_hir_typeck::check::check_fn
  23: rustc_hir_typeck::typeck_with_inspect::{closure#0}
  24: rustc_query_impl::query_impl::typeck::invoke_provider_fn::__rust_begin_short_backtrace
  25: rustc_query_impl::execution::try_execute_query::<rustc_data_structures::vec_cache::VecCache<rustc_span::def_id::LocalDefId, rustc_middle::query::erase::ErasedData<[u8; 8]>, rustc_middle::dep_graph::graph::DepNodeIndex>, true>
  26: rustc_query_impl::query_impl::typeck::execute_query_incr::__rust_end_short_backtrace
  27: <rustc_middle::ty::context::TyCtxt>::par_hir_body_owners::<rustc_hir_analysis::check_crate::{closure#1}>::{closure#0}
  28: rustc_hir_analysis::check_crate
  29: rustc_interface::passes::analysis
  30: rustc_query_impl::execution::try_execute_query::<rustc_middle::query::caches::SingleCache<rustc_middle::query::erase::ErasedData<[u8; 0]>>, true>
  31: rustc_query_impl::query_impl::analysis::execute_query_incr::__rust_end_short_backtrace
  32: rustc_interface::interface::run_compiler::<(), rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}
  33: std::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface::util::run_in_thread_with_globals<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<(), rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  34: <std::thread::lifecycle::spawn_unchecked<rustc_interface::util::run_in_thread_with_globals<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<(), rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  35: <std::sys::thread::unix::Thread>::new::thread_start
  36: <unknown>
  37: <unknown>

rustc version: 1.96.0-nightly (fb27476aa 2026-03-28)
platform: x86_64-unknown-linux-gnu
</pre>
</details>

<details><summary>Release build (llvm)</summary>
<pre>
delayed bug: Missing value for constant, but no error reported?
   0: <rustc_errors::DiagCtxtInner>::emit_diagnostic
   1: <rustc_errors::DiagCtxtHandle>::emit_diagnostic
   2: <rustc_span::ErrorGuaranteed as rustc_errors::diagnostic::EmissionGuarantee>::emit_producing_guarantee
   3: <rustc_errors::DiagCtxtHandle>::span_delayed_bug::<rustc_span::span_encoding::Span, &str>
   4: rustc_trait_selection::traits::const_evaluatable::is_const_evaluatable.cold
   5: <rustc_trait_selection::traits::select::SelectionContext>::evaluate_root_obligation
   6: rustc_traits::evaluate_obligation::evaluate_obligation
   7: rustc_query_impl::query_impl::evaluate_obligation::invoke_provider_fn::__rust_begin_short_backtrace
   8: rustc_query_impl::execution::try_execute_query::<rustc_middle::query::caches::DefaultCache<rustc_type_ir::canonical::CanonicalQueryInput<rustc_middle::ty::context::TyCtxt, rustc_middle::ty::ParamEnvAnd<rustc_middle::ty::predicate::Predicate>>, rustc_middle::query::erase::ErasedData<[u8; 2]>>, false>
   9: rustc_query_impl::query_impl::evaluate_obligation::execute_query_non_incr::__rust_end_short_backtrace
  10: <rustc_infer::infer::InferCtxt as rustc_trait_selection::traits::query::evaluate_obligation::InferCtxtExt>::evaluate_obligation_no_overflow
  11: <core::iter::adapters::map::Map<core::iter::adapters::filter::Filter<core::slice::iter::Iter<rustc_hir_typeck::method::probe::Candidate>, <rustc_hir_typeck::method::probe::ProbeContext>::consider_candidates::{closure#0}>, <rustc_hir_typeck::method::probe::ProbeContext>::consider_candidates::{closure#1}> as core::iter::traits::iterator::Iterator>::try_fold::<(), core::iter::traits::iterator::Iterator::find::check<(&rustc_hir_typeck::method::probe::Candidate, rustc_hir_typeck::method::probe::ProbeResult), &mut <rustc_hir_typeck::method::probe::ProbeContext>::consider_candidates::{closure#2}>::{closure#0}, core::ops::control_flow::ControlFlow<(&rustc_hir_typeck::method::probe::Candidate, rustc_hir_typeck::method::probe::ProbeResult)>>
  12: <rustc_hir_typeck::method::probe::ProbeContext>::pick_method
  13: <rustc_hir_typeck::method::probe::ProbeContext>::pick_all_method
  14: <rustc_hir_typeck::fn_ctxt::FnCtxt>::lookup_probe
  15: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_method_call
  16: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  17: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_overloaded_binop
  18: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_binop
  19: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  20: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_block
  21: <rustc_hir_typeck::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  22: rustc_hir_typeck::check::check_fn
  23: rustc_hir_typeck::typeck_with_inspect::{closure#0}
  24: rustc_query_impl::query_impl::typeck::invoke_provider_fn::__rust_begin_short_backtrace
  25: rustc_query_impl::execution::try_execute_query::<rustc_data_structures::vec_cache::VecCache<rustc_span::def_id::LocalDefId, rustc_middle::query::erase::ErasedData<[u8; 8]>, rustc_middle::dep_graph::graph::DepNodeIndex>, false>
  26: rustc_query_impl::query_impl::typeck::execute_query_non_incr::__rust_end_short_backtrace
  27: <rustc_middle::ty::context::TyCtxt>::par_hir_body_owners::<rustc_hir_analysis::check_crate::{closure#1}>::{closure#0}
  28: rustc_hir_analysis::check_crate
  29: rustc_interface::passes::analysis
  30: rustc_query_impl::execution::try_execute_query::<rustc_middle::query::caches::SingleCache<rustc_middle::query::erase::ErasedData<[u8; 0]>>, false>
  31: rustc_query_impl::query_impl::analysis::execute_query_non_incr::__rust_end_short_backtrace
  32: rustc_interface::interface::run_compiler::<(), rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}
  33: std::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface::util::run_in_thread_with_globals<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<(), rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  34: <std::thread::lifecycle::spawn_unchecked<rustc_interface::util::run_in_thread_with_globals<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<(), rustc_driver_impl::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  35: <std::sys::thread::unix::Thread>::new::thread_start
  36: <unknown>
  37: <unknown>

rustc version: 1.96.0-nightly (fb27476aa 2026-03-28)
platform: x86_64-unknown-linux-gnu
</pre>

</p>
</details>
</details>

Unstable features enabled: `read_array` & `generic_const_exprs`.

Happens with both debug builds with cranelift and release builds with llvm.
